tritium-registry/src/main/java/com/palantir/tritium/metrics/registry/TagMap.java

```diff
@@ -141,6 +148,14 @@ TagMap withEntry(String key, String value) {
         return new TagMap(newArray);
     }
 
+    TagMap withEntries(SortedMap<@Safe String, @Safe String> stringStringSortedMap) {
+        TagMap tagMap = this;
+        for (Map.Entry<String, String> entry : stringStringSortedMap.entrySet()) {
+            tagMap = tagMap.withEntry(entry.getKey(), entry.getValue());
```

Comment:
this is not optimized and will likely do `O(n)` expansions based on size of entries, so we'd want to optimize this